vortex-array/src/arrays/list/compute/is_constant.rs

```diff
@@ -0,0 +1,134 @@
+use vortex_error::VortexResult;
+
+use crate::arrays::{ListArray, ListVTable};
+use crate::compute::{IsConstantKernel, IsConstantKernelAdapter, IsConstantOpts};
+use crate::register_kernel;
+
+impl IsConstantKernel for ListVTable {
+    fn is_constant(&self, array: &ListArray, opts: &IsConstantOpts) -> VortexResult<Option<bool>> {
+        // At this point, we're guaranteed:
+        // - Array has at least 2 elements
+        // - All elements are valid (no nulls)
+
+        if opts.is_negligible_cost() {
+            return Ok(None);
+        }
+
+        let first_list_len = array.offset_at(1) - array.offset_at(0);
+        for i in 1..array.len() {
+            let current_list_len = array.offset_at(i + 1) - array.offset_at(i);
+            if current_list_len != first_list_len {
+                return Ok(Some(false));
+            }
+        }
+
+        // If all lists have the same length, compare the actual list contents
+        let first_scalar = array.scalar_at(0)?;
```

Comment:
Comparing list scalars is quite expensive... but this is where we would want some sort of `reduce_by(ALL_EQ)` compute function / operator...